### PR TITLE
refactor(connector): [Iatapay] remove default case handling

### PR DESCRIPTION
**Pull request by HeetVekariya**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [X] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

- Fix issue https://github.com/juspay/hyperswitch/issues/2272
- Changes to be made : Remove default case handling
- Added all the ```ConnectorAuthType``` (which are not ```auth_type``` ) against the default case
- File **changed: transformers.rs** (crates/router/src/connector/iatapay/transformers.rs)

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Comment by SanchithHegde** [comment]
@HeetVekariya Can you please address the failing CI checks?

You need to run `cargo +nightly fmt` for formatting code, and `cargo clippy --all-features --all-targets` to catch compilation errors and common mistakes. Refer to the [Cargo commands](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md#cargo-commands) section on the contributing guidelines for more info.

**Comment by HeetVekariya** [comment]
- @SanchithHegde I have done ```cargo +nightly fmt```, but while running this command ```cargo clippy --all-features --all-targets``` i got this:
image

- This changes are not done by me, as you can see in the images, So what should i do ?

**Comment by SanchithHegde** [comment]
Hey @HeetVekariya, as far as I understand, the `ConnectorAuthType` enum is not included under the scope of handling default cases. The enums that must be handled are primarily the payment methods related enums such as `PaymentMethodData` and `PaymentMethod`. You can also handle default cases for the `IatapayPaymentStatus` enum in your case.

And to clarify your question:

> This changes are not done by me, as you can see in the images, So what should i do ?

Here, `cargo` is informing you that you haven't handled one variant (`NoKey`) of the `ConnectorAuthType` enum in the `crates/router/src/connector/iatapay/transformers.rs` file (check the first half of the compilation logs in your screenshot). In the later part of the compilation logs, it just shows where the enum variant has been defined, for your reference. Need not necessarily mean that there's an error in that file.

**Comment by HeetVekariya** [comment]
- @SanchithHegde My bad, sorry for that. I found that i forget to include the ```NoKey``` against the default case, which causes the error.

**Review by prasunna09 — COMMENTED** [review]
Hi @HeetVekariya, Handling wild card match arms for AuthType is not included in the scope of the issue. Only Payment method data and payment method is. Can you please make the changes accordingly?

**Comment by HeetVekariya** [comment]
- @prasunna09 can you provide me more details, because i am not able to understand the suggested changes.

**Comment by HeetVekariya** [comment]
- @prasunna09 I have added ```PaymentMethod``` cases against default case.

File: crates/router/src/connector/iatapay/transformers.rs [hunk 1]
```diff
@@ -139,7 +150,14 @@ impl TryFrom<&types::ConnectorAuthType> for IatapayAuthType {
...
             }),
-            _ => Err(errors::ConnectorError::FailedToObtainAuthType)?,
+
+            types::ConnectorAuthType::HeaderKey { .. }
+            | types::ConnectorAuthType::BodyKey { .. }
+            | types::ConnectorAuthType::MultiAuthKey { .. }
+            | types::ConnectorAuthType::CurrencyAuthKey { .. }
+            | types::ConnectorAuthType::NoKey { .. } => {
+                Err(errors::ConnectorError::FailedToObtainAuthType)?
+            }
```
Comment (prasunna09) [inline crates/router/src/connector/iatapay/transformers.rs:160]: Please revert back this change as handling wild card match arms for AuthType is not in the scope. Rest looks good to me!

**Review by prasunna09 — CHANGES_REQUESTED** [review]

File: crates/router/src/connector/iatapay/transformers.rs (hunk 1 above)
Comment (HeetVekariya) [reply crates/router/src/connector/iatapay/transformers.rs]: I have reverted back the unwanted changes

**Review by prasunna09 — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]
Looks good to me!

Thanks for the PR, @HeetVekariya!

**Comment by HeetVekariya** [comment]
@SanchithHegde Thank you for merging
@prasunna09 Thank you for being with me for the queries.

**Comment by swangi-kumari** [comment]
Hey @HeetVekariya ,
Thanks a bunch for all your contributions! We've got a little something for you to show our appreciation.  
Just take a moment to fill out this [form](https://forms.gle/yqbtjFuP6oxLaBKN8), and get ready for some awesome swag coming your way.  
Thanks!

**Pull request merged**